crates/agglayer-clock/src/time.rs

```diff
@@ -76,7 +76,7 @@ impl TimeClock {
 
         // Compute the current Block height and Epoch number
         let current_block = self.update_block_height();
-        let current_epoch = self.calculate_epoch_number(current_block);
+        let current_epoch = Self::calculate_epoch_number(current_block, *self.epoch_duration);
```

Comment:
I am a bit confused here. The epoch is calculated from block number even for time-based clock?

Reply:
The `ClockRef` is always calculating the epoch based on the agglayer blocks.
In the time-based clock, a "block" is basically one second.